numpy/lib/function_base.py

```diff
@@ -158,6 +292,14 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
     >>> np.sum(hist*np.diff(bin_edges))
     1.0
 
+    Automated Bin Selection Methods example, using 2 peak random data with 2000 points
+
+    >>> from matplotlib import pyplot as plt
+    >>> a = np.hstack((np.random.randn(1000), np.random.randn(1000) * 2 + 5))
```

Comment:
Use a `np.random.RandomState` instance to make documentation builds deterministic.

Also -- do we actually generate plots from docstrings in the numpy documentation? It's worth checking on this.